BudgetTracker.Data.EntityFramework/Converters/TransactionConverter.cs

```diff
@@ -0,0 +1,40 @@
+using BudgetTracker.Business.Transactions;
+using BudgetTracker.Data.EntityFramework.Models;
+using System;
+
+namespace BudgetTracker.Data.EntityFramework.Converters
+{
+    public class TransactionConverter
```

Comment:
So why do we need two different types of converters, are we not saving the same transaction that we have in the business layer. Also, feels like anti DRY to have basically the same public static methods. 

Maybe we should just put it in the common folder.

Reply:
Which 2 are you talking about? There's 2 Converter classes... 1 for the Business layer to convert to and from API message model to domain object. The other for the EntityFramework layer to convert to and from business model to schema model.

Reply:
But they appear to be doing basically the same thing.

Reply:
The difference is who uses them. The one in the data layer is only used by the data layer. It's hidden from the other layers. The business layer doesn't even know there is a conversion happening at all. That's to make sure that the business layer only interacts with the domain models, not the entity framework models.

However, the domain layer does need to do some conversion of it's own for something else completely unrelated (convert input messages to the domain models). So it does need a converter of it's own.

Either way, we need to have the different conversions between models. But not all conversions should be available to every one. The data layer is the only one that should convert between domain and data models. And the domain layer is the only one that should convert between input/output messages and domain models.